src/features/addUncheckAllToReplyToModal.ts

```diff
@@ -0,0 +1,66 @@
+import './addUncheckAllToReplyToModal.css';
+
+import {makeBTDModule} from '../types/btdCommonTypes';
+
+enum LABELS {
+  SELECT_ALL = 'Select All',
+  UNSELECT_ALL = 'Unselect All',
+}
+
+export const addUncheckAllToReplyToModal = makeBTDModule(({TD}) => {
+  const jsModalsContainer = document.querySelector('.js-modals-container') as HTMLDivElement;
+  /**
+   * Listen to modal that are appended to the modal container
+   *
+   * Idea is to check if it's the right modal, since it doesn't appear to have a uuid or other consistant identifier
+   */
+  const observer = new MutationObserver((mutationsList, observer) => {
+    if (mutationsList[0].addedNodes.length > 0) {
+      const addedNode = mutationsList[0].addedNodes[0];
+      const isReplyToModal =
+        document.querySelector('.js-reply-checkbox') instanceof HTMLInputElement;
+
+      if (isReplyToModal) {
+        const containerForNewCheckbox =
+          addedNode.childNodes[1].childNodes[5].childNodes[0].childNodes[1];
+
+        const label = document.createElement('label');
+        label.textContent = LABELS.UNSELECT_ALL;
+        label.setAttribute('for', 'uncheck_all_reply_to_modal_checkbox');
+        label.classList.add('hover--underline');
+
+        const checkbox = document.createElement('input');
+        checkbox.type = 'checkbox';
+        checkbox.checked = true;
+        checkbox.hidden = true;
+        checkbox.id = 'uncheck_all_reply_to_modal_checkbox';
+
+        checkbox.addEventListener('change', (e) => {
+          const selectAll = checkbox.checked;
+          label.textContent = selectAll ? LABELS.UNSELECT_ALL : LABELS.SELECT_ALL;
+          Array.from(
+            document.querySelectorAll(
+              '.js-reply-checkbox:not([disabled])'
+            ) as NodeListOf<HTMLInputElement>
+          ).forEach((input) => {
+            if (selectAll) {
+              if (!input.checked) input.click();
+            } else {
+              if (input.checked) input.click();
+            }
+          });
+        });
+
+        Object.assign(label.style, {
+          cursor: 'pointer',
+        });
+
+        const container = document.createElement('div');
+        container.append(label, checkbox);
+        containerForNewCheckbox.appendChild(container);
+      }
+    }
+  });
+
+  observer.observe(jsModalsContainer, {characterData: false, childList: true, attributes: false});
+});
```

Comment:
I realize this is unlikely but I prefer to be careful: as far as I can tell your code doesn't try to check for the presence of the label you create? Meaning that you _could_ end up in an infinite loop where your MutationObserver listens for mutations, to which your callback will react by adding a DOM element, which will trigger the callback, and so on. Could you add some checks to prevent that?